### PR TITLE
ARTEMIS-2717 Isolate the current Quorum Vote implementation

### PR DESCRIPTION
**Pull request by franz1981**
This PR is part of the effort of providing a new pluggable QuorumVote implementation and aims to first isolate and refactor any "leak"  of the quorum vote logic outside `QuorumManager`.
This is not yet ready to be merged and is open to suggestions.

### CODE REVIEW

**Comment by franz1981** [comment]
@michaelandrepearce @brusdev 
In the current state, the only services exposed by `QuorumManager` are:

- `registerQuorum`/`unRegisterQuorum`: to register/unregister a listener for `nodeDown`/`nodeUP` events issued by `ClusterController::defaultLocator`. The high-level function, looking at `SharedNothingBackupQuorumm`, isto detect a change in the live role (ie live is going down)
- `isStillLive`: used by the live with `vote-on-replication-failure` to validate that is still live
- `hasLive`: used by the backup (on `SharedNothingBackupQuorumm`) to check, after received a disconnection failure or a `nodeDown` event (see `registerQuorum`/`unRegisterQuorum`), if there is any live in charge. 
- `handleQuorumVote`: to manage `QuorumVoteMessage` packets, perform a vote and send back a response. It's tied to `ClusterController` (again)

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/server/cluster/ClusterControl.java [hunk 1]
```diff
@@ -64,6 +62,10 @@ public ClusterControl(ClientSessionFactoryInternal sessionFactory, ActiveMQServe
       this.clusterPassword = server.getConfiguration().getClusterPassword();
    }
 
+   public Optional<Channel> getClusterChannel() {
```
Comment (franz1981) [inline artemis-server/src/main/java/org/apache/activemq/artemis/core/server/cluster/ClusterControl.java:65]: @michaelandrepearce @brusdev Not proud to have "exposed" the lower level `Channel` here, but it seems the simplest way to move away he many  `Quorum*` deps away from here

**Comment by brusdev** [comment]
Nice work. I have only one concern about `vote` used by `ColocatedActivation`.

**Comment by clebertsuconic** [comment]
I will merge this..

But please keep an eye on the testsuite.

**Pull request closed without merging**